validation/src/main/java/org/ehrbase/validation/terminology/ItemStructureVisitor.java

```diff
@@ -0,0 +1,424 @@
+package org.ehrbase.validation.terminology;
+
+
+import com.nedap.archie.rm.archetyped.Locatable;
+import com.nedap.archie.rm.changecontrol.OriginalVersion;
+import com.nedap.archie.rm.composition.*;
+
+import com.nedap.archie.rm.datastructures.*;
+import com.nedap.archie.rm.datavalues.DvCodedText;
+import com.nedap.archie.rm.datavalues.DvText;
+import com.nedap.archie.rm.datavalues.TermMapping;
+import com.nedap.archie.rm.datavalues.encapsulated.DvMultimedia;
+import com.nedap.archie.rm.datavalues.quantity.DvOrdered;
+import com.nedap.archie.rm.datavalues.quantity.DvOrdinal;
+import com.nedap.archie.rm.demographic.PartyRelationship;
+import com.nedap.archie.rm.ehr.EhrStatus;
+import com.nedap.archie.rm.generic.Participation;
+import com.nedap.archie.rm.integration.GenericEntry;
+import org.ehrbase.terminology.openehr.TerminologyInterface;
+import org.ehrbase.terminology.openehr.TerminologyService;
+import org.ehrbase.terminology.openehr.implementation.AttributeCodesetMapping;
+import org.ehrbase.terminology.openehr.implementation.LocalizedTerminologies;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Visitor of an item structure to perform specific validations:
+ * - CodePhrase depending on terminology
+ * - units
+ */
+public class ItemStructureVisitor implements I_ItemStructureVisitor {
+
+    protected static Logger log = LoggerFactory.getLogger(ItemStructureVisitor.class);
+    private int elementOccurrences = 0; //for statistics and testing
+    private ItemValidator itemValidator = new ItemValidator();
+    private LocalizedTerminologies localizedTerminologies;
+    private AttributeCodesetMapping codesetMapping;
+    private String itemStructureLanguage = "en"; //if a composition, the language can be found in the structure
+
+    public ItemStructureVisitor(LocalizedTerminologies localizedTerminologies) throws NoSuchMethodException, IllegalAccessException, ClassNotFoundException {
```

Comment:
I would be better to catch this Exceptions where there are occurring and wrap them in a Validation Exception. 
I also think this wrapper can be a RuntimeException.

Reply:
I had opened a ticket for this as I believe this is a wider change https://github.com/ehrbase/project_management/issues/84